airflow/models/taskinstance.py

```diff
@@ -1813,10 +1817,16 @@ class TaskInstance(Base, LoggingMixin):
     """
 
     __tablename__ = "task_instance"
-    task_id = Column(StringID(), primary_key=True, nullable=False)
-    dag_id = Column(StringID(), primary_key=True, nullable=False)
-    run_id = Column(StringID(), primary_key=True, nullable=False)
-    map_index = Column(Integer, primary_key=True, nullable=False, server_default=text("-1"))
+    id = Column(
+        String(36).with_variant(postgresql.UUID(as_uuid=False), "postgresql"),
+        primary_key=True,
+        default=lambda: str(uuid6.uuid7()),
```

Comment:
Declare as a top level function or classmethod so it can be used here and below (and easily swapped out for a different uuid7 implementation if needed)?

Reply:
Good point. Done in https://github.com/apache/airflow/pull/43243/commits/0f5dc0a0e496a38d53003ee9dc2941af0d8c9b74

Reply:
Maybe we don't need to force it to a string but instead add the SQLA typeadapter https://github.com/apache/airflow/blob/8a6b1fc4e0ab8ba42cc47d8390fcd62a41e924dd/airflow/utils/sqlalchemy.py#L48